File: Lista03_HerancaEPolimorfismo/Heranca/app/src/main/java/Lista03/ProdutoPerecivelEspecial.java
package Lista03;

import java.time.LocalDate;

public class ProdutoPerecivelEspecial extends ProdutoPerecivel{

    public ProdutoPerecivelEspecial(int codigo, double preco_unitario, String descricao, int quantidade_estoque, LocalDate data_validade) {
        super(codigo, preco_unitario, descricao, quantidade_estoque,data_validade);
    }

    public void imprimirNotaControle(){
        System.out.println("===NOTA DE CONTROLE===");
        System.out.println("Codigo: "+this.codigo); 
        System.out.println("Descricao: "+this.descricao); 
        System.out.println("Quantidade em estoque: "+this.quantidade_estoque); 
        System.out.println("Data de validade: "+this.data_validade); 
        System.out.println("======================");

        return;
    }  
}

File: Lista11_Python/Ex04_Heranca/produto_perecivel_especial.py
from produto import *
from produto_perecivel import *
import datetime

class ProdutoPerecivelEspecial(ProdutoPerecivel):

    def __str__(self):
        string = "Produto perecível especial:\n"
        string += super().__str__()
        string += ", validade: "+str(self._dataValidade.strftime("%d/%m/%Y"))
        return string
